### PR TITLE
Rename `case` to `ensemble`

### LINKED ISSUE #5902: Rename `case` to an agreed upon term
Ert contains a lot of references to 'case' because that was the unit of storage that existed before. 'Case' in Ert is equivalent to the modern concept of 'Ensemble'. However, for users, it may be that the modern 'experiment' concept is closer to how they perceive it.

We have started including `Experiment` in the gui, and as such should rename `Case` -> `Ensemble`

### PR DESCRIPTION
**Pull request by dafeda**
**Issue**
Resolves #5902 

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: tests/unit_tests/gui/conftest.py [hunk 1]
```diff
@@ -52,12 +52,12 @@
...
 def with_manage_tool(gui, qtbot: QtBot, callback) -> None:
     def handle_manage_dialog():
-        dialog = wait_for_child(gui, qtbot, ClosableDialog, name="manage-cases")
+        dialog = wait_for_child(gui, qtbot, ClosableDialog, name="manage-experiments")
         cases_panel = get_child(dialog, CaseInitializationConfigurationPanel)
         callback(dialog, cases_panel)
 
     QTimer.singleShot(1000, handle_manage_dialog)
-    manage_tool = gui.tools["Manage cases"]
+    manage_tool = gui.tools["Manage experiments"]
```
Comment (pinkwah) [inline tests/unit_tests/gui/conftest.py:60]: "Manage storage"?

**Review by pinkwah — APPROVED** [review]
Looks good. You didn't change all `sim_fs`es (Github isn't allowing me to comment on the spot).

I also think you can get rid of the deprecation warning. The CLI is meh as is.

File: tests/unit_tests/gui/conftest.py (hunk 1 above)
Comment (dafeda) [reply tests/unit_tests/gui/conftest.py:60]: Not sure we want users to think about "Storage" at all.
What do you think @oyvindeide  and @frode-aarstad ?

Comment (frode-aarstad) [reply tests/unit_tests/gui/conftest.py:60]: Not sure whats best. Is there a better word that can cover both experiments and ensembles? Eventually, we probably want the user to be able to delete experiments and ensembles in this tool aswell.

But "Manage experiments" is fine with me 🤷🏻

File: tests/unit_tests/gui/ertwidgets/test_caseselector.py [hunk 2]
```diff
@@ -1,77 +0,0 @@
-import pytest
```
Comment (frode-aarstad) [inline tests/unit_tests/gui/ertwidgets/test_caseselector.py:1]: Are the tests in this deleted file moved somewhere else or are they not needed anymore?

File: src/ert/__main__.py [hunk 3]
```diff
@@ -191,6 +192,28 @@ def run_lint_wrapper(args: Namespace, _: ErtPluginManager) -> None:
     lint_file(args.config)
 
 
+class DeprecatedAction(argparse.Action):
```
Comment (frode-aarstad) [inline src/ert/__main__.py:195]: Tried running  `(py3.11) [faar@be-lx138036 ert]$ ert ensemble_smoother --target-case ff1 test-data/poly_example/poly.ert 
/usr/lib64/python3.11/argparse.py:1978: UserWarning: --target-case is deprecated and will be removed in future versions. Use --target-ensemble instead.
  action(self, namespace, argument_values, option_string)` The last line with action(...) is not really needed

Comment (dafeda) [reply src/ert/__main__.py:195]: Fixed by settings stacklevel=1:

```python
        warnings.warn(
            f"{option_string} is deprecated and will be removed in future versions. {alternative_msg}",
            stacklevel=1,
        )
```

File: tests/unit_tests/gui/ertwidgets/test_caseselector.py (hunk 2 above)
Comment (dafeda) [reply tests/unit_tests/gui/ertwidgets/test_caseselector.py:1]: Good catch, I accidentally removed it.

**Pull request merged**